bindings/java/src/main/com/apple/foundationdb/RangeQuery.java

```diff
@@ -54,9 +54,10 @@
 	private final boolean reverse;
 	private final StreamingMode streamingMode;
 	private final EventKeeper eventKeeper;
+	private final TargetBytesStrategy bytesStrategy;
 
 	RangeQuery(FDBTransaction transaction, boolean isSnapshot, KeySelector begin, KeySelector end, int rowLimit,
-			boolean reverse, StreamingMode streamingMode, EventKeeper eventKeeper) {
+			boolean reverse, StreamingMode streamingMode, TargetBytesStrategy bytesStrategy,EventKeeper eventKeeper) {
```

Comment:
This seems slightly clunky to me in that both `streamingMode` and `bytesStrategy` are defining the strategy for the byte limits, with one generally taking precedence over the other.

If we could combine this into a single argument where you specify one of the existing streaming modes or your own strategy, then that would be cleaner.